File: CPP/CPP OOP/Aggregation.cpp
// cpp - Aggregation
#include <iostream>
class Address
{
public:
    int pin;
    std::string city, country;
    void setData(int pin, std::string city, std::string country)
    {
        this->city = city;
        this->pin = pin;
        this->country = country;
    }
};
class Student
{
public:
    std::string name;
    int no;
    Address ad;
    Student(int no, std::string name, Address ad)
    {
        this->name = name;
        this->no = no;
        this->ad = ad;
    }
    void Display()
    {
        std::cout << no << ' ' << name << std::endl;
        std::cout << ad.country << "\t" << ad.city << ' ' << ad.pin << std::endl;
    }
};
int main()
{
    Address ad;
    ad.setData(43, "Mumbai", "India");
    Student s(3, "MySelf", ad);
    s.Display();
    return 0;
}

File: Java/Java OOP/Aggregation.java
// java - Aggregation
public class Aggregation {
    public static void main(String[] args) {
        Address a = new Address(43, "Mumbai", "India");
        Student s = new Student("MySelf", 03);
        s.getAddress(a);
        s.DisplayData();
    }
}

class Student {
    String Name;
    int RollNO;
    Address add;

    Student(String Name, int RollNO) {
        this.Name = Name;
        this.RollNO = RollNO;
    }

    void DisplayData() {
        System.out.println(RollNO + " " + Name);
    }

    void getAddress(Address a) {
        this.add = a;
        System.out.println("Address Saved : ");
        System.out.println(a.Country + "\n" + a.City + " " + a.pin);
    }
}

class Address {
    int pin;
    String City;
    String Country;

    Address(int pin, String City, String Country) {
        this.City = City;
        this.Country = Country;
        this.pin = pin;
    }

    void DisplayData() {
        System.out.println(Country + "\n" + City + " " + pin);
    }
}